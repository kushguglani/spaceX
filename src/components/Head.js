import React from 'react';
import { Helmet } from 'react-helmet';

const Head = () => {
  return (
    <Helmet>
      <title>SpaceX Launch Programs</title>

      <meta charSet="UTF-8" />
      <meta
        name="viewport"
        content="width=device-width, initial-scale=1, shrink-to-fit=no"
      />
      <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
      <meta
        name="description"
        content="React SpaceX app with server side rendering"
      />

      <link
        rel="shortcut icon"
        // href={`${process.env.PUBLIC_URL}/favicon.ico?v1`}
        href="https://images2.imgbox.com/ff/81/EOWojaSj_o.png"
      />
    </Helmet>
  );
};

export default Head;
