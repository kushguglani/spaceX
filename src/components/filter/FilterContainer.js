import React from 'react'

import LaunchYear from './LaunchYear';
import IsSuccesfulLaunch from './IsSuccesfulLaunch';
import IsSuccesfulLanding from './IsSuccesfulLanding';

export default function filterContainer(props) {
    return (
        <div className="box-container">
            <h3 className="filter-heading">Filters</h3>
            <LaunchYear
                filter={props.filter}
                setFilter={props.setFilter}
                data-testid="launchYear" />
            <IsSuccesfulLaunch
                text="Successfull Launch"
                filter={props.filter}
                setFilter={props.setFilter}
                data-testid="successLaunch" />
            <IsSuccesfulLanding
                text="Successfull Landing"
                filter={props.filter}
                setFilter={props.setFilter}
                data-testid="successLand" />
        </div>
    )
}
