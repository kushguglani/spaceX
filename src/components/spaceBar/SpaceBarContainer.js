import React from 'react';

import SpaceBar from './SpaceBar';

const SpaceBarContainer = (props) => {
  const data = props.spacex.length > 0 ? props.spacex.map(space => {
    const { flight_number, links, rocket, mission_id, launch_year, launch_success } = space;
    return <SpaceBar
      flight_number={flight_number}
      rocket_name={rocket.rocket_name}
      mission_id={mission_id}
      launch_year={launch_year}
      launch_success={launch_success}
      img={links.mission_patch}
      key={space.flight_number}
    />
    // <div className="box-container space-card" key={space.flight_number}>
    //   <img src={space.links.mission_patch} alt={space.rocket.rocket_name + " image"} />
    //   <p className="rocket-name">{space.rocket.rocket_name}</p>
    //   <p>Mission Ids:
    // <span>{space.mission_id.length > 0 ? space.mission_id.map(curr =>
    //     <ul><li>{curr}</li></ul>) : 'No missions'}</span></p>
    //   <p>Luanch Year:<span> {space.launch_year}</span></p>
    //   <p>Successfull Luanch: <span>{"" + space.launch_success} </span></p>
    //   <p>Successfull Landing:<span> {"" + space.rocket.first_stage.cores[0].land_success}</span></p>
    // </div>
  }) : 'No SpaceX Found';
  return (
    <div className="space-container">
      {data}
    </div>
  );
};

export default SpaceBarContainer;
