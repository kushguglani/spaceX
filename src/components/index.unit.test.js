import React from 'react';
import { render } from '@testing-library/react';
import HeaderModule from './header/Header';
import FooterModule from './footer/Footer';
import FilterContainer from './filter/FilterContainer';
import SpaceBarContainer from './spaceBar/SpaceBarContainer';
import Head from './Head';

describe('Module', () => {
  it('matches header snapshot', () => {
    const { container } = render(<HeaderModule />);
    expect(container.cloneNode(true)).toMatchSnapshot();
  });
  it('matches FooterModule snapshot', () => {
    const { container } = render(<FooterModule />);
    expect(container.cloneNode(true)).toMatchSnapshot();
  });
  it('matches Head snapshot', () => {
    const { container } = render(<Head />);
    expect(container.cloneNode(true)).toMatchSnapshot();
  });
  it('matches FilterContainer snapshot', () => {
    const filter = {
      selectedYear: '',
      successfullLaunch: '',
      successfullLanding: '',
    };
    const { container } = render(<FilterContainer filter={filter} />);
    expect(container.cloneNode(true)).toMatchSnapshot();
  });
  it('matches SpaceBarContainer snapshot', () => {
    const space = {
      flight_number: 14,
      mission_name: 'CRS-3',
      mission_id: ['EE86F74'],
      launch_year: '2014',
      launch_window: 0,
      rocket: {
        rocket_id: 'falcon9',
        rocket_name: 'Falcon 9',
        rocket_type: 'v1.1',
      },
      launch_success: true,
      links: {
        mission_patch: 'https://images2.imgbox.com/ff/81/EOWojaSj_o.png',
      },
    };
    const { container } = render(<SpaceBarContainer spacex={space} />);
    expect(container.cloneNode(true)).toMatchSnapshot();
  });
});
