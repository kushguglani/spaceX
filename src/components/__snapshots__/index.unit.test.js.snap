// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module matches FilterContainer snapshot 1`] = `
<div>
  <div
    class="box-container"
  >
    <h3
      class="filter-heading"
    >
      Filters
    </h3>
    <div>
      <p
        class="center-text filter-text"
      >
        Luanch Year
      </p>
      <div
        class="btn-container"
      >
        <button
          class=""
        >
          2006
        </button>
        <button
          class=""
        >
          2007
        </button>
        <button
          class=""
        >
          2008
        </button>
        <button
          class=""
        >
          2009
        </button>
        <button
          class=""
        >
          2010
        </button>
        <button
          class=""
        >
          2011
        </button>
        <button
          class=""
        >
          2012
        </button>
        <button
          class=""
        >
          2013
        </button>
        <button
          class=""
        >
          2014
        </button>
        <button
          class=""
        >
          2015
        </button>
        <button
          class=""
        >
          2016
        </button>
        <button
          class=""
        >
          2017
        </button>
        <button
          class=""
        >
          2018
        </button>
        <button
          class=""
        >
          2019
        </button>
        <button
          class=""
        >
          2020
        </button>
      </div>
    </div>
    <div
      class="filter-box"
    >
      <p
        class="center-text filter-text"
      >
        Successfull Launch
      </p>
      <button
        class=""
      >
        True
      </button>
      <button
        class=""
      >
        False
      </button>
    </div>
    <div
      class="filter-box"
    >
      <p
        class="center-text filter-text"
      >
        Successfull Landing
      </p>
      <button
        class=""
      >
        True
      </button>
      <button
        class=""
      >
        False
      </button>
    </div>
  </div>
</div>
`;

exports[`Module matches FooterModule snapshot 1`] = `
<div>
  <h4
    class="center-text"
  >
    Developed by: 
    <em>
      Kush Guglani
    </em>
  </h4>
</div>
`;

exports[`Module matches Head snapshot 1`] = `<div />`;

exports[`Module matches SpaceBarContainer snapshot 1`] = `
<div>
  <div
    class="space-container"
  >
    No SpaceX Found
  </div>
</div>
`;

exports[`Module matches header snapshot 1`] = `
<div>
  <h3
    class="header"
  >
    SpaceX Launch Programs
  </h3>
</div>
`;
